### ISSUE TITLE
💅 `noUnresolvedImports` emits diagnostics for node.js builtins

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#8849]
### Environment information

<details>

```bash
playground

```

</details>

### Rule name

lint/nursery/noUnresolvedImports

### Playground link

https://biomejs.dev/playground/?lintRules=noUnresolvedImports&code=YwBvAG4AcwB0ACAAcABhAHQAaAAgAD0AIAByAGUAcQB1AGkAcgBlACgAIgBuAG8AZABlADoAcABhAHQAaAAiACkAOwA%3D

### Expected result

I would expect the rule to not flag this case.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
